src/common/buffer.cc

```diff
@@ -1309,7 +1309,8 @@ static ceph::spinlock debug_lock;
 	if (unlikely(raw && !raw->is_shareable())) {
 	  auto* clone = ptr_node::copy_hypercombined(*curbuf);
 	  curbuf = bl._buffers.erase_after_and_dispose(curbuf_prev);
-	  bl._buffers.insert_after(curbuf_prev++, *clone);
```

Comment:
Pushing the `curbuf_prev` iterator forward wasn't *sequenced-after` the insertion.